modules/dxp/apps/saml/saml-opensaml-integration/src/test/java/com/liferay/saml/opensaml/integration/internal/metadata/MetadataGeneratorTest.java

```diff
@@ -112,32 +92,45 @@ public void testMetadataGeneratorOmitsBlacklistedAlgorithms()
 				}
 			).build());
 
-		EntityDescriptor entityDescriptor =
-			metadataManagerImpl.getEntityDescriptor(
-				getMockHttpServletRequest(
-					"http://localhost:8080/c/portal/saml/metadata"));
+		Assert.assertFalse(
+			_checkMatch(
+				metadataManagerImpl.getEntityDescriptor(
+					getMockHttpServletRequest(
+						"http://localhost:8080/c/portal/saml/metadata"))));
+	}
 
+	private boolean _checkMatch(EntityDescriptor entityDescriptor) {
 		List<RoleDescriptor> roleDescriptors =
 			entityDescriptor.getRoleDescriptors();
 
-		Stream<RoleDescriptor> stream = roleDescriptors.stream();
+		List<XMLObject> xmlObjects = new ArrayList<>();
 
-		Assert.assertFalse(
-			stream.map(
-				RoleDescriptor::getExtensions
-			).map(
-				ElementExtensibleXMLObject::getUnknownXMLObjects
-			).flatMap(
-				Collection::stream
-			).filter(
-				SigningMethod.class::isInstance
-			).map(
-				SigningMethod.class::cast
-			).map(
-				SigningMethod::getAlgorithm
-			).anyMatch(
-				"http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"::equals
-			));
+		for (RoleDescriptor roleDescriptor : roleDescriptors) {
+			Extensions extensions = roleDescriptor.getExtensions();
+
+			xmlObjects.addAll(extensions.getUnknownXMLObjects());
+		}
+
+		boolean algorithmMatch = false;
+
+		for (XMLObject xmlObject : xmlObjects) {
+			if (!SigningMethod.class.isInstance(xmlObject)) {
+				continue;
+			}
+
+			SigningMethod signingMethod = (SigningMethod)xmlObject;
+
+			if (StringUtil.equals(
+					signingMethod.getAlgorithm(),
+					"http://www.w3.org/2001/04/xmldsig-more#rsa-sha256")) {
+
+				algorithmMatch = true;
+
+				break;
```

Comment:
@jeffwu0724 You can return true here, return false at the end of the method, and remove the boolean flag variable.